HW2/P5/P5.txt

```diff
@@ -0,0 +1,83 @@
+==============
+Multithreading
+==============
+Sample simulation frames per second with 1 thread:
+10.838947091 simulation frames per second
+10.0332119739 simulation frames per second
+10.0480180918 simulation frames per second
+9.95283542706 simulation frames per second
+8.52478288175 simulation frames per second
+
+Sample simulation frames per second with 4 threads:
+15.9815885935 simulation frames per second
+16.5032618532 simulation frames per second
+13.8779005324 simulation frames per second
+16.9721238705 simulation frames per second
+17.8785336743 simulation frames per second
+
+As shown above, the number of simulation frames per second is higher with 4 threads relative to 1 thread.
+However, the speedup is around 1.6x.
+
+=================================================================
+Spatial Decomposition with a Grid (Collision in 3 Adjacent Grids)
+=================================================================
+Sample simulation frames per second with 1 thread:
+263.170364421 simulation frames per second
+258.759107426 simulation frames per second
+255.524323007 simulation frames per second
+230.91725239 simulation frames per second
+260.40794702 simulation frames per second
+
+Sample simulation frames per second with 4 threads:
+475.39494994 simulation frames per second
+510.438802761 simulation frames per second
```

Comment:
I'm surprised that the grid didn't make an even bigger difference, maybe my comment on your double for loops partly explains why (that would account for about a 4x speedup)